src/adapters/ui/iconview.js

```diff
@@ -322,11 +316,13 @@ export class DesktopIconView extends EventEmitter {
           .then(entries => actions.setEntries(entries));
       }
 
-    }, view(fileIcon, themeIcon, droppable), this.$root);
+    }, view(this.core, fileIcon, themeIcon, droppable), this.$root);
```

Comment:
Instead of passing `core` here you can make a HOF below `const read =` with this and pass that instead:

```js
const computeLabel = createLabelComputer(this.core)
```

And then you can change the `computeLabel` to:

```js
const createLabelComputer = (core) => (entry) => {
  const [metadata] = core.make('osjs/packages').getPackages(pkg => (pkg.name === entry.filename));
  return entry.label || (metadata ? core.make('osjs/locale').translatableFlat(metadata.title) : entry.filename);
};
```

Which means that in your view you can just do this:

```js
const view = (computeLabel, fileIcon, themeIcon, droppable) => (state, actions) =>
```

```js
computeLabel(entry)
```

Reply:
Actually, if you wanna get fancy with it:

```js
const createLabelComputer = (core) => {
  const pkgs = f => core.make('osjs/packages').getPackages(f);
  const translate = n => core.make('osjs/locale').translatableFlat(n);

  return (entry) => {
    const [metadata] = pkgs(pkg => (pkg.name === entry.filename));
    return entry.label || (metadata ? translate(metadata.title) : entry.filename);
  };
};
```

Reply:
I like yours better, commit inbound, lol.

Reply:
Beware that I've edited this comment a few times, so make sure to refresh so that it's not out of date. I had some minor changes there :)

Reply:
Appreciate the heads up! Caught the 'title' property too, lol.